packages/core/addon/models/table.js

```diff
@@ -77,22 +78,63 @@ export default VisualizationBase.extend(Validations, {
   }
 });
 
+/**
+ * returns default dimension fields (show clause)
+ * @param {Object} dimension - dimension from request
+ * @returns {Object} - list of field names
+ */
+function getDefaultDimensionFields(dimension) {
+  let fields = get(dimension, 'dimension.fields');
+
+  if (isArray(fields)) {
+    return fields.filter(field => isArray(field.tags) && field.tags.includes('show')).map(field => field.name);
+  }
+
+  return [];
+}
+
+/**
+ * builds a single dimension column
+ * @param {Object} dimension - dimension from request
+ * @param {Object} column
+ * @param {String} fieldName
+ * @returns {Object} - dimension column
+ */
+function dimensionColumn(dimension, column, fieldName) {
+  let defaultName = dimensionFormat(get(dimension, 'dimension.longName'), fieldName);
+
+  return {
+    field: { dimension: get(dimension, 'dimension.name') },
+    type: 'dimension',
+    displayName: column ? column.displayName : defaultName
+  };
+}
+
 /**
  * builds dimension columns for new visualization builds
  * @param {Array} dimensions - dimensions from request
  * @param {Object} columnIndex - column lookup table indexed by dimension/metric name
  * @returns {Array} - list of dimensions columns
  */
 function buildDimensionColumns(dimensions, columnIndex) {
-  return dimensions.map(dimension => {
-    let column = columnIndex[get(dimension, 'dimension.name')];
-
-    return {
-      field: { dimension: get(dimension, 'dimension.name') },
-      type: 'dimension',
-      displayName: column ? column.displayName : get(dimension, 'dimension.longName')
-    };
+  let dimensionColumns = dimensions.map(dimension => {
+    let defaultFields = getDefaultDimensionFields(dimension),
+      dimensionName = get(dimension, 'dimension.name');
+
+    if (!defaultFields.length) {
+      let column = columnIndex[dimensionName];
+      return dimensionColumn(dimension, column);
+    } else {
+      //return an array of columns for each dimension field
+      return defaultFields.map(fieldName => {
+        let column = columnIndex[canonicalizeDimension(dimensionName, fieldName)];
+        return merge({}, { field: { show: fieldName } }, dimensionColumn(dimension, column, fieldName));
```

Comment:
I am not sure about the property `show`. `show` sounds like a boolean. `show` in fili is a metadata entity and a mechinism for fetching dimension `field`s. We want to model and describe the entity itself not the mechinism. 

You wouldn't say `Give me the device with the id and description shows`. Rather you would say `Give me the device with the id and description fields`.

Could we make the shape something like:
```js
{ 
   type: 'dimension',
   dimension: dimensionObject,
   field: 'id',
   displayName: 'Device (name)'
}

Reply:
I agree, also didn't see a reason for `field: {dimension: ... }}`

Reply:
`dimensionObject` is actually the dimension id, do we need an object here?

Reply:
or maybe
```
{
  type: 'dimension or metric',
  [dimension or metric]: {
    name: 'advanceTrafficCurrency',
    parameters: {} // for metric
    field: 'id', // for dimension - or should it be at the root level?
  }
  format: '' // for metric
  displayName: 'Traffic Currency'
}
```

Reply:
@jkusa @cythrawll @mattgraham1995 @balajimanoharan feel free to comment

Reply:
Yeah I wouldn't mind switching the shape so that we can use `type`, and `dimension`. freeing up `field` for this. it would be a lot cleaner.    I think I favor @daniellevinson last comment.

Reply:
@jkusa I like it. In any case it might be a breakable change

Reply:
@jkusa @cythrawll I've temporarily changed `show` to `field` which makes more semantical sense, so for now for dimension we have ```{
  type: 'dimension',
  field: {
    dimension: 'advanceTrafficCurrency',
    field: 'id'
  }
  displayName: 'Platform (id)'
}```
the more I'm diving into the shape changing the more I'm starting to think maybe we should separate it to its own PR, as this one might get a little messy since we rely on the `field` key for all column types, existing reports have `format` at the root level, change calls to `canonicalizeMetric`, update tests etc.

Reply:
I agree on probably making it a separate PR just for the breaking change. as you're right, it might get hairy. 

I'm fine having the subfield as `field` for the short term.